atsign-foundation/at_server
# fix: APKAM revoke and list changes

**- What I did**
- Add `revoked` operation in the `EnrollStatus` .
- Verify if an enrollment is authorized to perform operation. If yes, allow the operation else throw `UnAuthorizedException`.
- Add `revoke` functionality in the `enroll_verb_handler.dart`
- Refactor the `enroll_verb_handler.dart` as follows:
  - Move the logic of enrollment request, approval/deny/revoke into private method.

**- How to verify it**
- Added unit tests enrollment list operations
- Unit tests to verify if an enrollment is authorized to perform operations
- Added negative tests to ensure the enrollment approval/deny/revoke cannot be performed on an unauthenticated connection
- Added functional tests related to enrollment revoke functionality

**- Description for the changelog**
- Add revoke operation to enroll verb
